pkg/processor/k8sprocessor/pod_association.go

```diff
@@ -53,8 +58,8 @@ func extractPodID(ctx context.Context, attrs pcommon.Map, associations []kube.As
 			podIdentifierValue = kube.PodIdentifier(hostname)
 			return
 		}
-		podIdentifierKey = ""
-		return
+
+		return "", "", errors.New("pod association not set, could not assign other pod id")
```

Comment:
I decided to clean some returns in this function, because they were inconsistent in which other. Basing on all return calls in this function:
 - if an error is returned, use `return "", "", error"
 - if everything is fine, assign a value to `podIdentifierValue` (and `podIdentifierKey` if applies) and use `return`